### PR TITLE
Upgrade to Polkadot v0.9.37

### PR DESCRIPTION
**Pull request by jasl**
Because of Cumulus not released 0.9.34 and 0.9.35 so we upgrade from 0.9.33 to 0.9.37

https://github.com/paritytech/polkadot/releases/tag/v0.9.34
https://github.com/paritytech/polkadot/releases/tag/v0.9.35
https://github.com/paritytech/polkadot/releases/tag/v0.9.36
https://github.com/paritytech/polkadot/releases/tag/v0.9.37

https://github.com/paritytech/substrate/compare/polkadot-v0.9.33...polkadot-v0.9.34
https://github.com/paritytech/substrate/compare/polkadot-v0.9.34...polkadot-v0.9.35
https://github.com/paritytech/substrate/compare/polkadot-v0.9.35...polkadot-v0.9.36
https://github.com/paritytech/substrate/compare/polkadot-v0.9.36...polkadot-v0.9.37

Breaking changes:
- https://github.com/paritytech/substrate/pull/12891

Notable changes:
- https://github.com/paritytech/substrate/pull/12310
- https://github.com/paritytech/substrate/pull/12775
- https://github.com/paritytech/substrate/pull/12813
- https://github.com/paritytech/substrate/pull/12844
- https://github.com/paritytech/substrate/pull/13045
- https://github.com/paritytech/substrate/pull/13066
- https://github.com/paritytech/substrate/pull/13080

In addition:
- Sync Phala to https://github.com/Phala-Network/phala-blockchain/commit/c5056e01edc0a8747cf6164742bd3636de3cfaef
- Use Rust nightly-2023-02-03

Because of we need to run migrations for assets and balances in this version, I think we should upgrade Phala parachain as well

We need test migration carefully

~~TODO~~:
- Evaluate that should we bump tx version? @h4x3rotab 
  - https://github.com/paritytech/cumulus/issues/2017#issuecomment-1361582387
  - For Phala, we skip too much versions and we bumped for Khala before, should we do for Phala?
- Evaluate assets pallet migration @tolak 
  - https://github.com/paritytech/cumulus/pull/1742
- Evaluate balances pallet migration @tolak 
  - https://github.com/paritytech/cumulus/pull/1943
  - https://github.com/paritytech/substrate/pull/13009
  - the migration trigger again on Statemint on 0.9.36 https://github.com/paritytech/cumulus/pull/2031
- (not blocking) wating reply from Parity about `--state-pruning` issue

### CODE REVIEW

**Comment by jasl** [comment]
NOTE:

```
test memdb::tests::consolidate ... FAILED
test memdb::tests::memorydb_remove_and_purge ... FAILED

failures:

---- memdb::tests::consolidate stdout ----
thread 'memdb::tests::consolidate' panicked at 'called `Option::unwrap()` on a `None` value', crates/phala-trie-storage/src/memdb.rs:579:56

---- memdb::tests::memorydb_remove_and_purge stdout ----
thread 'memdb::tests::memorydb_remove_and_purge' panicked at 'called `Option::unwrap()` on a `None` value', crates/phala-trie-storage/src/memdb.rs:547:52
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace


failures:
    memdb::tests::consolidate
    memdb::tests::memorydb_remove_and_purge

test result: FAILED. 1 passed; 2 failed; 0 ignored; 0 measured; 0 filtered out; finished in 0.00s
```

Relates to Phala repo

**Comment by h4x3rotab** [comment]
Re. tx version: Let's check how tx version affects the blockchain behavior in substrate. If it's only for external signer (e.g. hardware or mobile wallet) to distinguish changes in extrinsic args, we'd rather just keep it as is. However, if it affects the security of the blockchain, we have no choice but update it.

**Review by tolak — APPROVED** [review]
Go through the pallet-assets and SubBridge relevant changes, LGTM

**Comment by jasl** [comment]
I shall merge this, feel free to double check it

File: crates/phala-types/src/lib.rs [hunk 1]
```diff
@@ -62,7 +62,6 @@ pub mod messaging {
 
     #[derive(Encode, Decode, Debug, TypeInfo)]
     pub struct WorkerInfo {
-        pub attestation_provider: Option<AttestationProvider>,
```
[the remark this answers is not in the document]
Comment (jasl) [reply crates/phala-types/src/lib.rs:65]: Kevin says this removing is correct, or it will be problem for PRuntime V2

**Pull request merged**